cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go

```diff
@@ -35,42 +35,169 @@ const (
+	105: {description: "Microsoft Windows 7", importerOSIDs: []string{"windows-7-x64-byol", "windows-7-x86-byol"}},
+	106: {description: "CentOS 32-bit", importerOSIDs: []string{}},
+	107: {description: "CentOS 64-bit", importerOSIDs: []string{"centos-6", "centos-7", "centos-8"}},
+	108: {description: "Oracle Linux 32-bit", importerOSIDs: []string{}},
+	109: {description: "Oracle Linux 64-bit", importerOSIDs: []string{}},
+	110: {description: "eComStation 32-bitx", importerOSIDs: []string{}},
+	111: {description: "Microsoft Windows Server 2011", importerOSIDs: []string{}},
+	113: {description: "Microsoft Windows Server 2012", importerOSIDs: []string{"windows-2012", "windows-2012-byol"}},
+	114: {description: "Microsoft Windows 8", importerOSIDs: []string{"windows-8-x86-byol"}, nonDeterministic: true},
+	115: {description: "Microsoft Windows 8 64-bit", importerOSIDs: []string{"windows-8-x64-byol"}, nonDeterministic: true},
+	116: {description: "Microsoft Windows Server 2012 R2", importerOSIDs: []string{"windows-2012r2", "windows-2012r2-byol"}},
+	117: {description: "Microsoft Windows Server 2016", importerOSIDs: []string{"windows-2016", "windows-2016-byol"}},
+	118: {description: "Microsoft Windows 8.1", importerOSIDs: []string{}},
+	119: {description: "Microsoft Windows 8.1 64-bit", importerOSIDs: []string{}},
+	120: {description: "Microsoft Windows 10", importerOSIDs: []string{"windows-10-x86-byol"}, nonDeterministic: true},
+	121: {description: "Microsoft Windows 10 64-bit", importerOSIDs: []string{"windows-10-x64-byol"}, nonDeterministic: true},
+	//TODO: windows-2019, windows-2019-byol
+}
+
+//Mapping OVF osType attribute to importer OS ID
+//Some might have multiple importer OS ID  values. In that case, user need to provide a value via --os flag.
 // Some might have only one option but we can't select it automatically as we cannot guarantee
-// correctness. All Windows Client imports will be included here, to check that they
-// expect a Bring Your Own License import.
-var noMappingOSTypes = map[string][]string{
-	"ubuntuGuest":           {"ubuntu-1404"},
-	"ubuntu64Guest":         {"ubuntu-1404", "ubuntu-1604", "ubuntu-1804"},
-	"windows7Guest":         {"windows-7-x86-byol"},
-	"windows7_64Guest":      {"windows-7-x64-byol"},
-	"windows8Guest":         {"windows-8-x86-byol"},
-	"windows8_64Guest":      {"windows-8-x64-byol"},
-	"windows9Guest":         {"windows-10-x86-byol"},
-	"windows9_64Guest":      {"windows-10-x64-byol"},
-	"windows8Server64Guest": {"windows-2012", "windows-2012r2"},
-	"windows9Server64Guest": {"windows-2016", "windows-2019"},
+// correctness. All Windows Client imports are in this category due to the fact we can't assume BYOL licensing.
+//Full list: https://vdc-download.vmware.com/vmwb-repository/dcr-public/da47f910-60ac-438b-8b9b-6122f4d14524/16b7274a-bf8b-4b4c-a05e-746f2aa93c8c/doc/vim.vm.GuestOsDescriptor.GuestOsIdentifier.html
+var ovfOSTypeToOSID = map[string]osInfo{
```

Comment:
I noticed that we are missing some of the OS versions listed here that are on https://cloud.devsite.corp.google.com/compute/docs/images/os-details . Is that mostly because we have to do extra work to enable those new versions?

Reply:
The list of operating systems in OVF descriptor spec is not something we can add to. E.g. COS would probably be listed as "Other". I'm not sure this answers your question?

Reply:
I meant Debian 10 and similar newer versions of OSes. It seems to be listed in the link that you have provided in the comment.

Reply:
Image import (and OVF import which relies on image import) currently don't support Debian 10 import, even though it's supported by GCE.